File: Cryptography and security libraries and algorithms/C/Replit Ghostwriter/Hash generation.c
#include <openssl/evp.h>
#include <openssl/scrypt.h>
#include <string.h>

unsigned char *Hash_Pass_1(const char *password) {
    unsigned char *hash = malloc(EVP_MAX_MD_SIZE);
    EVP_MD_CTX *mdctx;
    mdctx = EVP_MD_CTX_create();
    EVP_DigestInit_ex(mdctx, EVP_md5(), NULL);
    EVP_DigestUpdate(mdctx, password, strlen(password));
    unsigned int hash_len;
    EVP_DigestFinal_ex(mdctx, hash, &hash_len);
    EVP_MD_CTX_destroy(mdctx);

    return hash;
}

unsigned char *Hash_Pass_2(const char *password) {
    unsigned char *hash = malloc(64);
    int ret = scrypt((const uint8_t *)password, strlen(password), NULL, 0, 1<<14, 8, 1, hash, 64);
    if (ret != 0) {
        free(hash);
        return NULL;
    }

    return hash;
}

int main() {
    const char *password = "123Pass";

    // generate the hash for MyPassword and store it in a variable called MyHash
    unsigned char *MyHash = Hash_Pass_1(password);

File: Cryptography and security libraries and algorithms/Python/Code complete/Hash generation.py
import hashlib
import scrypt
def Hash_Pass_1(password):
    NewHash=hashlib.md5(password)
    return(NewHash)
def Hash_Pass_2(password):
    NewHash=scrypt.hash(password,salt=None,n=2**14,r=8,p=1)
    return(NewHash)
MyPassword="123Pass"
#generate the hash for MyPassword and store it in a variable called MyHash
MyHash=Hash_Pass_1(MyPassword)
#print the hash to the screen
print("The MD5 Hash of MyPassword is:",MyHash.hexdigest())
#generate a new hash for MyPassword and store it in a variable called NewHash2
NewHash2=Hash_Pass_2(MyPassword)
#print the hash to the screen
print("The Scrypt Hash of MyPassword is:",NewHash2.hexdigest())
